hathor/transaction/base_transaction.py

```diff
@@ -887,6 +894,33 @@ def is_ready_for_validation(self) -> bool:
                 return False
         return True
 
+    @property
```

Comment:
If I recall correctly your reasoning for not liking `@property` is because it can misleadingly hide expensive operations by creating the impression that it is a simple access.

In this case, I argue that by design this _is_ the intended interpretation. The user should indeed consider this a simple access, because it is:

```python
    @property
    def static_metadata(self) -> StaticMetadataT:
        """Get this vertex's static metadata. Assumes it has been initialized."""
        assert self._static_metadata is not None
        return self._static_metadata
```

I could easily use `self._static_metadata` everywhere (it is a "real" property), but then I would have to `assert not None` everywhere. This `@property` is then a simple convenience for performing that assertion for us.

This is exactly the same pattern we currently use for `BaseTransaction.hash`, it is a `@property`:

https://github.com/HathorNetwork/hathor-core/blob/f1a56b7b75411f54dbe68e22875d544baf610765/hathor/transaction/base_transaction.py#L287-L290

I think using it is much better for caller ergonomics.

This also intentionally contrasts with our existing `BaseTransaction.get_metadata()`, which is a bloated method that does not performs only access, but also initializes metadata and even uses the storage. By design I want `static_metadata` to feel like normal vertex attributes that are always available. The [next PR](https://github.com/HathorNetwork/hathor-core/pull/1014) describes this idea a bit more:

> 1. All vertices are required to have static metadata set before they're saved to the storage, meaning when a vertex is retrieved from the storage, it's guaranteed to have static metadata. No need to perform updates and such.
> 2. Static metadata can be interpreted more as attributes that are as constant and present as intrinsic vertex attributes, and less like dynamic metadata attributes (like voided_by and validation).

Reply:
Got it. Makes sense and I feel it's a good use of properties.